### DISCUSSION TITLE
Insert Null Values

### DISCUSSION THREAD

**Discussion by cloud303-cholden** [SeaQL/sea-query#734] [Q&A] [answered]
Is it possible to insert null values? I saw in #100 that it might not be supported. I'm hoping to use sea-query with sqlx.

**Comment by tyt2y3** [comment, marked as answer]
Just a quick note, `Value`'s definition has since been changed to `Decimal(Option<Box<Decimal>>)`, so every variant is nullable.
Checkout the sqlx examples too! https://github.com/SeaQL/sea-query/tree/master/examples/sqlx_mysql

**Reply by cloud303-cholden** [reply]
Thanks for the response. I'm specifically dealing with a struct with an `Option<String>` field. From the docs it looks like calling `into()` on the value that's passed to `values()` should be sufficient to convert to the `String` variant of the `Value` enum you mentioned, but the error said a `String` was expected. I'll play around with it more and reopen this if needed. Thanks!

**Reply by orangesoup** [reply]
@cloud303-cholden Hey! I'm in the same situation currently, have you managed to solve it?

**Reply by cloud303-cholden** [reply]
@orangesoup Sorry I didn't end up having time for this, and I am currently using sea-query/sea-orm instead of sea-query/sqlx. Best of luck!

**Reply by Expurple** [reply]
@cloud303-cholden

> From the docs it looks like calling `into()` on the value that's passed to `values()` should be sufficient to convert to the `String` variant of the `Value` enum you mentioned

Yes, but [`values()`](https://docs.rs/sea-query/0.32.0/sea_query/query/struct.InsertStatement.html#method.values) accepts an `IntoIterator<Item = SimpleExpr>`, so you also need to convert `Value -> SimpleExpr`.

I think, in a future version SeaQuery could be improved to accept something like `IntoIterator<Item: Into<SimpleExpr>>` and do this for you automatically.
